File: C++/2972.cpp
#include <vector>
#include <algorithm>
class Solution {
public:
    long long countSubarrays(std::vector<int>& nums, int k) {
        int target = *std::max_element(nums.begin(), nums.end()), counter = 0, l = 0;
        long long ans = 0;
        for(auto r : nums){
            counter += r == target;
            while(counter == k) counter -= nums[l++] == target; 
            ans += l;  
        }  
        return ans;
    }
};

File: Python/2972.py
class Solution:
    def countSubarrays(self, nums: list[int], k: int) -> int:
        target = max(nums)
        counter = 0
        ans = 0
        l = 0
        
        for r in range(len(nums)):
            if nums[r] == target:
                counter += 1

            while (counter > k or (counter == k and nums[l] != target)) and l <= r:
                if nums[l] == target:
                    counter -= 1
                l += 1

            if counter == k:
                ans += l+1

        return ans
        